tests/lib/lazy_orm.py

```diff
@@ -0,0 +1,107 @@
+
+The solution I have implemented below is a "lazy ORM" class. These looks pretty much
+like regular ORM classes when you define them, and their instances act just enough like
+ORM objects that you can build up a set a fixtures with them; but it's not until you
+"hydrate" these objects that they become actual ORM instances. This gives us the
+opportunity to customise the hydration process for each database and produce ORM classes
+with the appropriate column types.
+
+This is, obviously, a terrible thing. But it's at least terrible in a sort of generic
+Python way rather than reaching deep into the internals of SQLAlchemy. So I felt on
+balance that it was the least-worst of the many terrible solutions I tried.
+
+It also has the advantage of being reasonably self-contained and easy to back out: the
+lazy ORM classes are just a find-and-replace away from being regular ORM classes; and
+the hydration class is used in only a single place.
+"""
+import functools
+
+import sqlalchemy
+
+
+class LazyBase:
+    def __init_subclass__(cls, **kwargs):
+        # Keep a record of defined columns
+        cls._columns = {}
+        for key, value in vars(cls).items():
+            assert not isinstance(
+                value, sqlalchemy.Column
+            ), f"'{key}' must be a LazyColumn not a sqlalchemy.Column"
+
+            if isinstance(value, LazyColumn):
+                cls._columns[key] = value
+                # If we don't do this, instances of the class will have the column
+                # definition as their default column value, which is not what we want
+                setattr(cls, key, None)
+
+    def __init__(self, **kwargs):
+        # This is almost verbatim what the default ORM constructor does
+        for key, value in kwargs.items():
+            if key not in self._columns:
+                raise TypeError(f"{key} is not a valid keyword argument")
+            setattr(self, key, value)
+
+
+class LazyColumn:
+    """
+    This does nothing apart from record the arguments to the column constructor so we
+    can use them to create an actual `sqlalchemy.Column` later (possibly after
+    customising the column definition to suit the target database).
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+
+
+class HydrateLazyObjects:
+    def __init__(self, type_map=None):
+        """
+        `type_map` if supplied should be a dictionary mapping SQLAlchemy column types to
+        alternative types
+        """
+        self.type_map = type_map or {}
+        self.orm_base_class = sqlalchemy.orm.declarative_base()
+
+    def __call__(self, item):
+        """
+        Takes an arbitrary object and, if it is an instance of LazyBase, will transform
+        it into an instance of an appropriate ORM class. Otherwise returns its argument
+        unchanged.
+        """
+        if isinstance(item, LazyBase):
+            orm_class = self.get_orm_class(item.__class__)
```

Comment:
This appears to be the wrong method name (should be `make_orm_class`).

Reply:
Yep! I did a quick "fix" just before leaving on Friday and then saw the test fail after it was too late to do anything about it.